### PR TITLE
[Refactor][Manifest] widen Softmax/LogSoftmax x.dtype to fp32 and flip to implemented

### LINKED ISSUE #1089: [Refactor][Manifest] widen Softmax/LogSoftmax dtype to match PyTorch (fp16|bf16|fp32)
### Symptom / Motivation

`SoftmaxFwdOp` and `LogSoftmaxFwdOp` were promoted to `status: implemented` in PR #1084 and reverted to `spec-only` in commit 3be5d00 due to a dtype-parity gap: op code and tests accept `torch.float32`, while the manifest declares `x: float16 | bfloat16` only.

Investigation shows the **manifest is the side that's wrong**, not the code:

- `torch.nn.functional.softmax` / `log_softmax` are PyTorch public APIs that support every floating dtype (fp16, bf16, fp32, fp64) on CUDA. fp32 is the most common input.
- `_softmax_base.py` docstrings already advertise "float32, float16, or bfloat16"; runtime only checks `x.dtype == self.dtype` with no per-dtype gating.
- `tests/ops/test_softmax.py` exercises fp32/fp16/bf16 in both smoke and full tiers and they pass.
- The narrowing was introduced in PR #1072 without a PyTorch-API justification.

Per `.claude/rules/manifest-trust-model.md`: when an op name matches a PyTorch op, the manifest signature must match PyTorch's public API. The current `float16 | bfloat16` declaration violates that rule. The fix is to **widen the manifest**, not narrow the implementation.

### Root Cause

PR #1072 narrowed `SoftmaxFwdOp` / `LogSoftmaxFwdOp` `x.dtype` to `float16 | bfloat16` without aligning to PyTorch's public API. The L1+ validator can't catch this because it has no notion of "must match `ref_api`'s dtype surface."

### Related Files

- `tileops/manifest/reduction.yaml` (entries `SoftmaxFwdOp`, `LogSoftmaxFwdOp`)
- `tileops/ops/reduction/softmax.py`
- `tileops/ops/reduction/log_softmax.py`
- `tileops/ops/reduction/_softmax_base.py`
- `tests/ops/test_softmax.py`
- `benchmarks/ops/bench_softmax.py`

## Goal

Align manifest dtype with PyTorch public API and re-flip both ops to `status: implemented`.

## Plan

Single migration PR, human-developer confirmed (no agent-only path):

1. In `tileops/manifest/reduction.yaml`, change `SoftmaxFwdOp.signature.inputs.x.dtype` and `LogSoftmaxFwdOp.signature.inputs.x.dtype` from `"float16 | bfloat16"` to `"float16 | bfloat16 | float32"`.
2. Add fp32 rows to the `workloads` lists for representative shapes (at least one attn-weights and one lm-head-logits entry per op) so benchmarks cover the widened surface.
3. Flip both entries to `status: implemented` and remove the inline dtype-parity gap-marker comment on the `status:` line.
4. Run `scripts/validate_manifest.py --check-op SoftmaxFwdOp` and `--check-op LogSoftmaxFwdOp`.
5. Run `pytest tests/ops/test_softmax.py -m "smoke or full"` and confirm fp32 rows are still green.

Note on fp64: PyTorch supports it, but TileOPs has no fp64 kernel path elsewhere and the existing softmax implementation has not been validated on fp64. Out of scope here; can be added later if a workload demands it.

## Constraints

- Manifest is source of truth and must match the PyTorch public API for `ref_api`-bearing ops (`.claude/rules/manifest-trust-model.md`).
- Do not touch op code, tests, or benchmarks — they are already correct. The implementation matches PyTorch; the manifest is the outlier. The migration is manifest-only plus the status flip.
- The existing `dtype=<torch.dtype>` PyTorch param remains BLOCKED on missing-manifest-capability (cannot express conditional output dtype in the DSL); not in scope.

## Acceptance Criteria

- [ ] AC-1: `tileops/manifest/reduction.yaml` declares `x: float16 | bfloat16 | float32` for both `SoftmaxFwdOp` and `LogSoftmaxFwdOp`, with at least one fp32 workload row each.
- [ ] AC-2: Both manifest entries show `status: implemented` with the dtype-parity gap-marker comment removed.
- [ ] AC-3: `scripts/validate_manifest.py --check-op SoftmaxFwdOp` and `--check-op LogSoftmaxFwdOp` pass.
- [ ] AC-4: `pytest tests/ops/test_softmax.py -m "smoke or full"` is green with no fp32 rows skipped/xfailed.

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #1089

## Summary

- Widen `SoftmaxFwdOp.signature.inputs.x.dtype` and `LogSoftmaxFwdOp.signature.inputs.x.dtype` from `float16 | bfloat16` to `float16 | bfloat16 | float32`, aligning the manifest with the PyTorch public API and with the existing implementation that already handles fp32.
- Add fp32 workload rows (`attn-weights-4k-fp32` and `lm-head-logits-fp32`) for both ops, flip both entries to `status: implemented`, and remove the inline dtype-parity gap-marker comments.

## Test plan

- [x] AC-1: `reduction.yaml` declares `x: float16 | bfloat16 | float32` for both SoftmaxFwdOp and LogSoftmaxFwdOp; each has at least one fp32 workload row.
- [x] AC-2: Both manifest entries show `status: implemented` with the dtype-parity gap-marker comment removed.
- [x] AC-3: `scripts/validate_manifest.py --check-op SoftmaxFwdOp` and `--check-op LogSoftmaxFwdOp` both pass.
- [x] AC-4: `pytest tests/ops/test_softmax.py -m "smoke or full"` green — 126 passed, 0 skipped/xfailed.

## Regression

Manifest-only change. No op code, kernel, test, or benchmark touched. Test suite for softmax family remains green at 126/126 with the new fp32 rows exercised.

### CODE REVIEW

File: tileops/manifest/reduction.yaml [hunk 1]
```diff
@@ -7,11 +7,11 @@
 SoftmaxFwdOp:
   ref_api: "torch.nn.functional.softmax"
   family: reduction
-  status: spec-only  # impl/tests still accept float32; manifest x dtype is float16 | bfloat16 only. Flip to implemented when dtype parity matches.
+  status: implemented
```
Comment (Ibuki-wind) [inline tileops/manifest/reduction.yaml]: `status: implemented` now claims `SoftmaxFwdOp` satisfies the manifest implemented contract, but `torch.nn.functional.softmax` exposes `dim: Optional[int] = None` and `dtype: Optional[...] = None` while this entry still types `dim` as `int` and omits `dtype`; R2/R15 require PyTorch-supported params and semantics for implemented entries -> keep this `spec-only` until the manifest can express and the op/tests cover `dim=None` and `dtype`, or add the missing parameters/behavior before flipping the status.

File: tileops/manifest/reduction.yaml [hunk 2]
```diff
@@ -55,11 +57,11 @@ SoftmaxFwdOp:
 LogSoftmaxFwdOp:
   ref_api: "torch.nn.functional.log_softmax"
   family: reduction
-  status: spec-only  # impl/tests still accept float32; manifest x dtype is float16 | bfloat16 only. Flip to implemented when dtype parity matches.
+  status: implemented
```
Comment (Ibuki-wind) [inline tileops/manifest/reduction.yaml]: `status: implemented` now claims `LogSoftmaxFwdOp` satisfies the manifest implemented contract, but `torch.nn.functional.log_softmax` exposes `dim: Optional[int] = None` and `dtype: Optional[...] = None` while this entry still types `dim` as `int` and omits `dtype`; R2/R15 require PyTorch-supported params and semantics for implemented entries -> keep this `spec-only` until the manifest can express and the op/tests cover `dim=None` and `dtype`, or add the missing parameters/behavior before flipping the status.

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
### Overall
The status flip is not ready; keep these entries non-implemented until the manifest matches the public API contract.

File: tileops/manifest/reduction.yaml (hunk 1 above)
Comment (lcy-seso) [reply tileops/manifest/reduction.yaml]: Accepted. Reverted `status: implemented` → `spec-only` for both `SoftmaxFwdOp` and `LogSoftmaxFwdOp` in 684b47c. Also widened `dim` to `int | None` (default `null`) so the manifest matches PyTorch's public signature, and added a tighter inline gap-marker on the `status:` line listing the two impl gaps (dim=None implicit-axis, dtype kwarg). `dtype` parameter still omitted — the manifest DSL cannot express conditional output dtype; tracked in the inline NOTE.

File: tileops/manifest/reduction.yaml (hunk 2 above)
Comment (lcy-seso) [reply tileops/manifest/reduction.yaml]: Accepted. Reverted `status: implemented` → `spec-only` for both `SoftmaxFwdOp` and `LogSoftmaxFwdOp` in 684b47c. Also widened `dim` to `int | None` (default `null`) so the manifest matches PyTorch's public signature, and added a tighter inline gap-marker on the `status:` line listing the two impl gaps (dim=None implicit-axis, dtype kwarg). `dtype` parameter still omitted — the manifest DSL cannot express conditional output dtype; tracked in the inline NOTE.

File: tileops/manifest/reduction.yaml [hunk 3]
```diff
@@ -33,6 +33,8 @@ SoftmaxFwdOp:
     - {x_shape: [32, 32, 32768], dtypes: [bfloat16], label: "attn-weights-32k"}
     # LM head logits (batch, vocab_size)
     - {x_shape: [4, 102400], dtypes: [float16, bfloat16], label: "lm-head-logits"}
+    - {x_shape: [32, 32, 4096], dtypes: [float32], label: "attn-weights-4k-fp32"}
+    - {x_shape: [4, 102400], dtypes: [float32], label: "lm-head-logits-fp32"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/reduction.yaml]: Applied. Regrouped fp32 workload rows next to their fp16/bfloat16 counterparts in 684b47c.

File: tileops/manifest/reduction.yaml [hunk 4]
```diff
@@ -81,6 +83,8 @@ LogSoftmaxFwdOp:
     - {x_shape: [32, 32, 32768], dtypes: [bfloat16], label: "attn-weights-32k"}
     # LM head logits
     - {x_shape: [4, 102400], dtypes: [float16, bfloat16], label: "lm-head-logits"}
+    - {x_shape: [32, 32, 4096], dtypes: [float32], label: "attn-weights-4k-fp32"}
+    - {x_shape: [4, 102400], dtypes: [float32], label: "lm-head-logits-fp32"}
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/reduction.yaml]: Applied. Regrouped fp32 workload rows next to their fp16/bfloat16 counterparts in 684b47c.

**Review by Ibuki-wind — APPROVED** [review]
### Overall
No blocking issues remain; the affected manifest entries are back to spec-only and targeted manifest validation passes with warnings only.

### Prior blockers
All 2 prior blockers resolved — see thread replies by reverting the affected entries to spec-only.
